test/apis/component_recognizer_test.go

```diff
@@ -105,7 +105,7 @@ func TestComponentDetectionOnOpenLiberty(t *testing.T) {
 }
 
 func TestComponentDetectionOnQuarkus(t *testing.T) {
-	isComponentsInProject(t, "quarkus", 1, "java", "code-with-quarkus-maven")
+	isComponentsInProject(t, "quarkus", 2, "java", "code-with-quarkus-maven")
```

Comment:
I see that now we have more components recognized for this test. Will this have an impact on other projects? IMO it improves the situation as now we are able to have a more detailed analysis. WDYT?

Reply:
In this case, the `quarkus` project will be detected as 2 components since it will look for inner container components.

The updated detection flow will now match the above cases except for case 2 -- in this case, it will find a container component and the python component which is expected

Reply:
IMO we shouldn't detect the dockerfile component as it is part of another component.